marklogic-rdf4j/src/main/java/com/marklogic/semantics/rdf4j/MarkLogicRepositoryConnection.java

```diff
@@ -676,7 +677,7 @@ public boolean hasStatement(Resource subject, IRI predicate, Value object, boole
             logger.debug(queryString);
             MarkLogicBooleanQuery query = prepareBooleanQuery(queryString); // baseuri ?
 
-            setBindings(query, subject, predicate, object, contexts);
+            setBindings(query, subject, predicate, Util.getInstance().skolemize(object), contexts);
```

Comment:
I believe, as long as the program is not rerun, it will skolemize to the same value, as per the hashCode() methods contract.